### DISCUSSION TITLE
Building fish from source - getting _fzf_preview_file command not found

### DISCUSSION THREAD

**Discussion by ristillu** [PatrickF1/fzf.fish#208] [Q&A] [answered]
I'm experimenting with migrating from bash to fish. Because of this I have not set fish as my default $SHELL and I therefore manually instantiate sessions starting from a bash session. I have been using fd and bat and fzf and so on for a long time with bash. I installed fisher and PatrickF1/fzf.fish successfully and can make them work. However each time I bring up a new session I have to jump through some hoops to make previewing work.

Relevant lines in my config.fish:
```sh
    # Setup for PatrickF1/fzf.fish
    set -Ux fzf_preview_dir_cmd ~/.cargo/bin/lsd --all --long --color always
    set -Ux fzf_preview_file_cmd ~/.cargo/bin/bat --color always --style full
    set -Ux fzf_fd_opts --hidden --follow --exclude=.git
    fzf_configure_bindings --directory=\ct
```
`<C-T>` is/was my mapping in bash for fzf completion

I type in a fish session `ls <C-T>` and it brings up an fzf overlay but I get `sh: _fzf_preview_file: command not found`. This is not correct - when this happens I can cancel out of the fzf and `ls ~/.config/fish/functions` and _fzf_preview_file.fish exists in the right place.

In order to make my previewing work I can do the following:
```sh
fisher update PatrickF1/fzf.fish
source ~/.config/fish/config.fish
```
but this is obviously a slow and expensive process due to the git fetch and I'd really prefer not to have to keep repeating an unnecessary step.

Can you tell me what I might be doing wrong? It feels almost like a dependency ordering problem? I'm so new to fish I don't know how to debug these sorts of things yet.

**Comment by PatrickF1** [comment]
Hi, I honestly don't even know where to begin debugging this. The steps you listed for making previewing work each time you load fish shouldn't have be run at all.

```
fisher update PatrickF1/fzf.fish
```
should permanently update the plugin

And fish should automatically be sourcing config.fish so you wouldn't need to run
```
source ~/.config/fish/config.fish
```

Are you using Nix or something where your shell always reverts to a locked state unless you change some configs?

**Comment by ristillu** [comment, marked as answer]
I'm using RHEL.

I solved the problem. I was building fish from source and had run `make` but not `sudo make install`, while I played with it and decided whether to take the leap. Installing must set up some bindings differently. Before when I was running `~/git/path/to/fish/repo/build/fish` manually/locally it must have been confused and when I then did the fisher stuff it created some bindings/lookups each time.

**Reply by PatrickF1** [reply]
Wow that's helpful! This may help future users as well!
Were you building from source to use HEAD?

**Comment by ristillu** [comment]
I build from source because I'm used to available packages not being very up to date for RHEL and because I generally have good luck with HEAD for different repos and am not afraid of tracking it for my personal environment setup. Although I just checked and saw I could easily have installed an up-to-date fish 3.3.1 RPM ...
